packages/react/src/components/date-picker/date-picker.tsx

```diff
@@ -464,6 +482,18 @@ export const Datepicker = forwardRef(({
         }
     }, [onChange]);
 
+    function handleClickOutside(event: MouseEvent<HTMLInputElement>): void {
+        if (
+            calendarRef.current
+            && dateInputRef.current
+            && calendarButtonRef.current
+            && !calendarRef.current.contains(event.target as Node)
+            && !calendarButtonRef.current.contains(event.target as Node)
+        ) {
+            dateInputRef.current.setOpen(false);
+        }
```

Comment:
Ah et je pense qu'avec ma suggestion, pour garder le comportement qui ferme le DatePicker sur le click du bouton, il faudrait changer ceci pour
```suggestion
        if (
            dateInputRef.current
            && calendarButtonRef.current
            && eventIsInside(event as unknown as Event, calendarButtonRef.current)
        ) {
            dateInputRef.current.setOpen(true);
        }
```
`eventIsInside` c'est une fonction qui est nécessaire pour gérer la shadow dom. Malheureusement on dirait qu'il y a un fuck de type avec l'event de React et l'event natif, alors faut faire `event as unknown as Event`.

Reply:
Très fort! Beaucoup plus clean de même. C'est exactement ce qui me manquait! 
`event as unknown as Event` est probablement un des bouts de code les plus weird que j'ai vu lol